File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = prices[0]
        profit = 0
        for i in range(1, len(prices)):
            if prices[i] < minPrice:
                minPrice = prices[i]
            
            profit = max(profit, prices[i]-minPrice )
        return profit









        # minPrice = prices[0]
        # maxProfit =0
        # for i in range(1,len(prices)):
        #     if prices[i]<minPrice:
        #         minPrice = prices[i]
        
        # #     else:
        # #         profit = prices[i]-minPrice
        # #         maxProfit = max(profit, maxProfit)
        # # return maxProfit

        #     profit = prices[i]-minPrice
        #     if profit>maxProfit:
        #         maxProfit = profit

        #     # maxProfit = max(maxProfit, prices[i]-minPrice)

        # return maxProfit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int min = prices[0];
        for(int i = 1;i<prices.length;i++){
            if (min>prices[i]){
                min = prices[i];
            }
            
            profit = Math.max(profit, prices[i]-min);
            // System.out.println(profit);
        }
        return profit;
    }
}